python/cudf/dataframe/string.py

```diff
@@ -127,9 +127,80 @@ def cat(self, others=None, sep=None, na_rep=None):
             (same type as caller) of str dtype is returned.
         """
         from cudf.dataframe import Series, Index
+
+        def is_list_like(others):
+            from collections.abc import Sequence
+            if isinstance(others, (Sequence,)) and \
+                    not isinstance(others, (str, bytes)):
+                return True
+            else:
+                return False
+
         if isinstance(others, (Series, Index)):
+            '''
+            If others is just another Series/Index,
+            great go ahead with concatenation
+            '''
             assert others.dtype == np.dtype('object')
             others = others.data
+        elif is_list_like(others) and others:
+            '''
+            If others is a list-like object (in our case lists & tuples)
+            just another Series/Index, great go ahead with concatenation.
+            '''
+
+            '''
+            Picking first element and checking if it really adheres to
+            list like conditions, if not we don't have the
+            necessity to scan the rest of the list.
+            '''
+            first = others[0]
```

Comment:
There will be no index error at this place because of `others` truth check which is chained with `and`. 
Reason I have chained it post `is_list_like(others)` is we won't be able to do a truth check like `if pd.Series:` as that will be ambiguous and results in ValueError. Since `is_list_like` will guard such cases before hand not letting it to pass till the next chained item in and expression this statement works well with all supported types we pass.